src/protagonist/Engine/Ingest/Image/ImageServer/Clients/CantaloupeThumbsClient.cs

```diff
@@ -0,0 +1,86 @@
+﻿using DLCS.Core.Exceptions;
+using DLCS.Core.FileSystem;
+using DLCS.Core.Strings;
+using DLCS.Core.Types;
+using DLCS.Model.Templates;
+using Engine.Ingest.Image.ImageServer.Manipulation;
+using Engine.Settings;
+using Microsoft.Extensions.Options;
+
+namespace Engine.Ingest.Image.ImageServer.Clients;
+
+public class CantaloupeThumbsClient : ICantaloupeThumbsClient
+{
+    private readonly HttpClient thumbsClient;
+    private readonly EngineSettings engineSettings;
+    private readonly IFileSystem fileSystem;
+    private readonly IImageManipulator imageManipulator;
+        
+    public CantaloupeThumbsClient(
+        HttpClient thumbsClient,
```

Comment:
`CantaloupeClient`? Could be confused with the thumbnail service?

Reply:
I think I put it down as `CantaloupeThumbsClient` to avoid confusion over it being used to complete image processing as well as thumbs - though it might be best to change this to `CantaloupeClient` and change the method to `GenerateThumbs`, or similar

Reply:
Sorry, I should have clarified - I was just meaning `HttpClient thumbsClient`.